cmd/cstorvolumeclaim/cstorvolumeclaim.go

```diff
@@ -0,0 +1,450 @@
+		"openebs.io/persistent-volume":   claim.Name,
+		"openebs.io/version":             version.GetVersion(),
+	}
+}
+
+// getTargetServiceSelectors get the selectors for cstor volume service
+func getTargetServiceSelectors(claim *apis.CStorVolumeClaim) map[string]string {
+	return map[string]string{
+		"app":                          "cstor-volume-manager",
+		"openebs.io/target":            "cstor-target",
+		"openebs.io/persistent-volume": claim.Name,
+	}
+}
+
+// getTargetServiceOwnerReference get the ownerReference for cstorvolume service
+func getTargetServiceOwnerReference(claim *apis.CStorVolumeClaim) []metav1.OwnerReference {
+	return []metav1.OwnerReference{
+		*metav1.NewControllerRef(claim,
+			apis.SchemeGroupVersion.WithKind(cvcKind)),
+	}
+}
+
+// getCVRLabels get the labels for cstorvolumereplica
+func getCVRLabels(pool *apis.CStorPool, volumeName string) map[string]string {
+	return map[string]string{
+		"cstorpool.openebs.io/name":    pool.Name,
+		"cstorpool.openebs.io/uid":     string(pool.UID),
+		"cstorvolume.openebs.io/name":  volumeName,
+		"openebs.io/persistent-volume": volumeName,
+		"openebs.io/version":           version.GetVersion(),
+	}
+}
+
+// getCVRAnnotations get the annotations for cstorvolumereplica
+func getCVRAnnotations(pool *apis.CStorPool) map[string]string {
+	return map[string]string{
+		"cstorpool.openebs.io/hostname": pool.Labels["kubernetes.io/hostname"],
+	}
+}
+
+// getCVRFinalizer get the finalizer for cstorvolumereplica
+func getCVRFinalizer() []string {
+	return []string{
+		CStorVolumeReplicaFinalizer,
+	}
+}
+
+// getCVROwnerReference get the ownerReference for cstorvolumereplica
+func getCVROwnerReference(cv *apis.CStorVolume) []metav1.OwnerReference {
+	return []metav1.OwnerReference{
+		*metav1.NewControllerRef(cv,
+			apis.SchemeGroupVersion.WithKind(cvKind)),
+	}
+}
+
+// getCVLabels get the labels for cstorvolume
+func getCVLabels(claim *apis.CStorVolumeClaim) map[string]string {
+	return map[string]string{
+		"openebs.io/persistent-volume": claim.Name,
+		"openebs.io/version":           version.GetVersion(),
+	}
+}
+
+// getCVOwnerReference get the ownerReference for cstorvolume
+func getCVOwnerReference(cvc *apis.CStorVolumeClaim) []metav1.OwnerReference {
+	return []metav1.OwnerReference{
+		*metav1.NewControllerRef(cvc,
+			apis.SchemeGroupVersion.WithKind(cvcKind)),
+	}
+}
+
+// getStorageClass return storageclass object for a given storageClass Name.
+// or error if any.
+func getStorageClass(
+	scName string,
+) (*storagev1.StorageClass, error) {
+	if scName == "" {
+		return nil, errors.New("failed to get storageclass: name missing")
+	}
+	scObj, err := sc.NewKubeClient().Get(scName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"failed to get storageclass {%s}",
+			scName,
+		)
+	}
+	return scObj, nil
+}
+
+// getReplicationFactor gets the ReplicationFactor from the from given storageclass
+func getReplicationFactor(
```

Comment:
Are these functions still required -- if the replicaCount is obtained from CStorVolumeClaim ?

Reply:
I will clean this up in upcoming PR's.. if that's fine?